proposals/collection-literals.md

```diff
@@ -555,6 +513,107 @@ Not having a *known length* does not prevent any result from being created. Howe
 
         The counts passed to `CreateArray` are used to provide a starting size hint to prevent wasteful resizes.
 
+## Dictionaries
+
+[*Constructible collection types*](#constructible-collection-types) is updated to include dictionaries.
+
+The following are _constructible collection types_ that can be used to construct collection literals.
+
+* ...
+* `Dictionary<TKey, TValue>`
+* Interface types `I<TKey, TValue>` implemented by `Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
+
+_Note that `Dictionary<TKey, TValue>` does not include a public `Add()` instance method with a single parameter, so the type is not constructible as a *collection initializer*._
+
+_Use `this[int index] { set; }` rather than `Add()` when constructing a dictionary for consistent set rather than add semantics._
+
+```c#
+var d = [x:x1, y:y1]; // {{x, x1}, {y, y1}}
+d = [..d, y:y2];      // {{x, x1}, {y, y2}}
+d = [x:x2, ..d];      // {{x, x2}, {y, y2}}
+```
+
+The [*natural type*](#natural-type) is updated as follows.
+
+If the [*natural element type*](#natural-element-type-updated) `T` can be determined and it is some `KeyValuePair<TKey, TValue>`, then the *natural type* of the collection is `Dictionary<TKey, TValue>`; otherwise, the *natural type* of the collection is `List<T>`.
```

Comment:
ok. i'm fine with this, though we likely should have a note in the higher up natural type section that mentions that the algorithm will be tweaked later.